File: smoothie-sample/src/main/java/com/example/smoothie/deps/PresenterContextNamer.java
package com.example.smoothie.deps;

import android.app.Application;
import com.example.smoothie.PersistActivity;
import javax.inject.Inject;

@PersistActivity.Presenter
public class PresenterContextNamer {
  private static int countInstances = 0;

  @Inject Application application;

  public PresenterContextNamer() {
    countInstances++;
  }

  public String getInstanceCount() {
    return "Instance# " + countInstances;
  }
}

File: smoothie-sample/src/main/java/com/example/smoothie/kotlin/deps/PresenterContextNamer.kt
package com.example.smoothie.kotlin.deps

import com.example.smoothie.kotlin.PersistActivity
import javax.inject.Inject

@PersistActivity.Presenter
class PresenterContextNamer @Inject constructor() {

    init {
        countInstances++
    }

    val instanceCount: String
        get() = "Instance# $countInstances"

    companion object {
        private var countInstances = 0
    }
}
